platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -809,6 +848,18 @@ public boolean onFling(final float posX, final float posY, final float velocityX
                 }
                 return true;
             }
+
+            @Override
+            public boolean onCancelFling() {
+                if (responder == null || !responder.onCancelFling()) {
+                    if (panGesturesEnabled) {
+                        cancelCameraAnimation();
+                        // TODO: Ideally should call onRegionDidChange if map state "InChanging" - VT(09/10/2018)
+                        mapChangeListener.onRegionDidChange(true);
```

Comment:
This doesn't seem right - it results in the listener receiving `onRegionDidChange` as the first event whenever a touch begins. Is this what your comment refers to?

Reply:
Yes, thats exactly what my comment refers to.

`onRegionDidChange` should only be called if map was "changing"! So until that is implemented (WIP), we either have to do this or ignore `onRegionDidChange` for fling cancel!